hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+from hs_file_types.models import GenericLogicalFile
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
```

Comment:
@hyi  I am not sure we have reached a decision to store metadata using key/value (HStore) for all existing resource types. It would be lot of work to change the existing code. What I am suggesting that  we may try key/value for the netcdf file type and see how that works. Then we can make a decision if we want to use this approach for all file types.  In that case there will be some code rewriting for raster file type. The only benefit of using key/value for metadata is that there won't be database migration issues as it relates to metadata. It's is less likely that we will end up with less code if we use key/value. We will have to write more custom validation code in case of key/value.